internal/tfschema/tfschema.go

```diff
@@ -110,41 +108,42 @@ func getTerragruntPluginPath(dir string) string {
 	return ret
 }
 
-func getClient(providerName string, dir string, iacType common.IACType) (tfschema.Client, error) {
+func getResourceSchema(resourceType string, dir string, iacType common.IACType) (*ResourceSchema, error) {
 	if iacType == common.Terragrunt {
 		// which mode of terragrunt it is (with or without cache folder).
 		if _, err := os.Stat(dir + "/.terragrunt-cache"); err == nil {
 			dir = getTerragruntPluginPath(dir)
 		}
 	}
 
-	providerToClientMapLock.Lock()
-	defer providerToClientMapLock.Unlock()
+	providerSchemasMapLock.Lock()
+	defer providerSchemasMapLock.Unlock()
 
-	client, exists := providerToClientMap[providerName]
-	if exists {
-		return client, nil
-	}
+	providerSchemas, ok := providerSchemasMap[dir]
```

Comment:
The provider schemas are cached based on the directory location.
This code has been completely changed.

If the provider schemas are not cached there's a "newer" command `terraform providers schema -json` that extracts the schemas.